assertTrue for scala 3 doesn't support package qualified identifiers

An assertion like 

```scala
assertTrue(zio.duration.Duration.fromNanos(1000) == zio.duration.Duration.Zero)
```

is translated into invalid code that looks basically like this:

```scala
zio.test.Arrow.succeed[zio](zio)
    .>>>[zio.duration](zio.test.Arrow.fromFunction[zio, zio.duration](((from: zio) => from.duration)))
    .>>>[zio.duration.package.type](zio.test.Arrow.fromFunction[zio.duration, zio.duration.package.type]((from: zio.duration) => from))
    .>>>[zio.duration.Duration.type](zio.test.Arrow.fromFunction[zio.duration.package.type, zio.duration.Duration.type]((from: zio.duration.package.type) => from.Duration))
    .>>>[java.time.Duration](zio.test.Arrow.fromFunction[zio.duration.Duration.type, java.time.Duration]((from: zio.duration.Duration.type) => from.fromNanos(1000L)))
    .>>>[scala.Boolean](zio.test.Assertions.equalTo[java.time.Duration](zio.duration.Duration.Zero)(zio.test.OptionalImplicit.none[zio.test.Diff[java.time.Duration]]))
```

the scalajs version fails to compile with
`java.lang.AssertionError: assertion failed: Cannot use package as value: Select(This(Ident(<root>)),zio)`

while the jvm version fails at runtime with
`java.lang.NoClassDefFoundError: zio`

the test that contains the assertion is currently green on the jvm because it is annotated with `@@ failing`.

I will try to make a PR for this.

--- a/test-tests/shared/src/test/scala/zio/test/SmartAssertionSpec.scala
+++ b/test-tests/shared/src/test/scala/zio/test/SmartAssertionSpec.scala
@@ -285,6 +285,9 @@ object SmartAssertionSpec extends ZIOBaseSpec {
         val string = "Sunday Everyday"
         assertTrue(string == "Saturday Todays")
       } @@ failing
-    )
+    ),
+    test("Package qualified identifiers") {
+      assertTrue(zio.duration.Duration.fromNanos(0) == zio.duration.Duration.Zero)
+    }
   )
 }

--- a/test/shared/src/main/scala-dotty/zio/test/Macros.scala
+++ b/test/shared/src/main/scala-dotty/zio/test/Macros.scala
@@ -103,17 +103,22 @@ class SmartAssertMacros(ctx: Quotes)  {
             case (tpeArgs, None) => Select.overloaded(param, name, tpeArgs, List.empty)
           }
 
-        lhs.tpe.widen.asType match {
-          case '[l] =>
-            val selectBody = '{
-              (from: l) => ${ body('{from}.asTerm).asExprOf[A] }
-            }
-            val lhsExpr = transform(lhs.asExprOf[l]).asExprOf[Arrow[Any, l]]
-            val assertExpr = '{Arrow.fromFunction[l, A](${selectBody})}
-            val pos = summon[PositionContext]
-            val span = Expr((lhs.pos.end - pos.start, method.pos.end - pos.start))
-            '{$lhsExpr >>> $assertExpr.span($span)}
-        }
+        val tpe = lhs.tpe.widen
+
+        if(tpe.typeSymbol.isPackageDef)
+          '{Arrow.succeed($expr).span(${method.span})}
+        else
+          tpe.asType match {
+            case '[l] =>
+              val selectBody = '{
+                (from: l) => ${ body('{from}.asTerm).asExprOf[A] }
+              }
+              val lhsExpr = transform(lhs.asExprOf[l]).asExprOf[Arrow[Any, l]]
+              val assertExpr = '{Arrow.fromFunction[l, A](${selectBody})}
+              val pos = summon[PositionContext]
+              val span = Expr((lhs.pos.end - pos.start, method.pos.end - pos.start))
+              '{$lhsExpr >>> $assertExpr.span($span)}
+          }
 
 
       case Unseal(tree) =>

